test/vault/Vault.t.sol

```diff
@@ -91,612 +80,534 @@ contract VaultTest is Test, NoIsolate, GasSnapshot {
-        // router => vault.lock
-        // vault.lock => periphery.lockAcquired
-        // periphery.lockAcquired => FakePoolManager.XXX => vault.accountAppBalanceDelta
-
-        vault.sync(currency0);
-        vault.sync(currency1);
-        currency0.transfer(address(vault), 10 ether);
+    function _testLockNotSettledWithoutPayment() external {
+        poolManager1.mockAccounting(poolKey1, -10 ether, -10 ether);
+    }
 
+    function testLockNotSettledWithoutFullyPayment() public noIsolate {
         vm.expectRevert(IVault.CurrencyNotSettled.selector);
-        vm.prank(address(fakePoolManagerRouter));
-        vault.lock(hex"02");
+        vault.lock(abi.encodeCall(VaultTest._testLockNotSettledWithoutFullyPayment, ()));
     }
 
-    function testLockNotSettled3() public noIsolate {
-        // router => vault.lock
-        // vault.lock => periphery.lockAcquired
-        // periphery.lockAcquired => FakePoolManager.XXX => vault.accountAppBalanceDelta
+    function _testLockNotSettledWithoutFullyPayment() external {
+        poolManager1.mockAccounting(poolKey1, -10 ether, -10 ether);
 
-        vault.sync(currency0);
+        currency0.settle(vault, address(this), 10 ether, false);
+
+        // didnt actually transfer the currency
         vault.sync(currency1);
-        currency0.transfer(address(vault), 10 ether);
-        currency1.transfer(address(vault), 8 ether);
+        vault.settle();
+    }
 
+    function testLockNotSettledAsPayTooMuch() public noIsolate {
         vm.expectRevert(IVault.CurrencyNotSettled.selector);
-        vm.prank(address(fakePoolManagerRouter));
-        vault.lock(hex"02");
+        vault.lock(abi.encodeCall(VaultTest._testLockNotSettledAsPayTooMuch, ()));
     }
 
-    function testLockNotSettled4() public noIsolate {
-        // router => vault.lock
-        // vault.lock => periphery.lockAcquired
-        // periphery.lockAcquired => FakePoolManager.XXX => vault.accountAppBalanceDelta
-
-        vault.sync(currency0);
-        vault.sync(currency1);
-        currency0.transfer(address(vault), 10 ether);
-        currency1.transfer(address(vault), 12 ether);
-
-        vm.expectRevert(IVault.CurrencyNotSettled.selector);
-        vm.prank(address(fakePoolManagerRouter));
-        vault.lock(hex"02");
+    function _testLockNotSettledAsPayTooMuch() external {
+        poolManager1.mockAccounting(poolKey1, -10 ether, -10 ether);
+        currency0.settle(vault, address(this), 10 ether, false);
+        currency1.settle(vault, address(this), 12 ether, false);
     }
 
     function testNotCorrectPoolManager() public {
-        // router => vault.lock
-        // vault.lock => periphery.lockAcquired
-        // periphery.lockAcquired => FakePoolManager.XXX => vault.accountAppBalanceDelta
+        // DOUBLE-CHECK:
+        // The tx will complete without revert, is this going to be a problem ?
+        vault.lock(abi.encodeCall(VaultTest._testNotCorrectPoolManager, ()));
+    }
```

Comment:
Could u guys take a look at this ? It turns out there won't be any error when `poolManager` specified in poolKey is not matching `msg.sender`.  Personally, i don't see any potential harm this could bring

Reply:
This case will only happened in invalid pool manager.
In normal cases , pool manager will check "pool.checkPoolInitialized();" , 
if you passed a pool manager with zero address , can not pass this checker.

Reply:
Sure, will add comments to point out that poolManager should be responsible for the validation here